File: 0150. Evaluate Reverse Polish Notation/Solution0150.py
from typing import List


class Solution0150:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for token in tokens:
            if token not in "+-*/":
                stack.append(int(token))
                continue
                
            num2 = stack.pop()            
            num1 = stack.pop()
            tmp = 0
            if token == "+":
                tmp = num1 + num2
            elif token == "-":
                tmp = num1 - num2
            elif token == "*":
                tmp = num1 * num2
            else:
                tmp = int(num1 / num2)
                
            stack.append(tmp)
            
        return stack.pop()

File: 0150. Evaluate Reverse Polish Notation/Solution0150.js
var evalRPN = function(tokens) {
    let stk = [];
    let ops = {
        '+': (a, b) => a + b,
        '-': (a, b) => a - b,
        '*': (a, b) => a * b,
        '/': (a, b) => a / b >= 0 ? Math.floor(a / b) : Math.ceil(a / b)
    };

    for (let t of tokens) {
        if (ops[t]) {
            let n2 = stk.pop();
            let n1 = stk.pop();
            stk.push(ops[t](n1, n2));
        } else {
            stk.push(Number(t));
        }
    }
    return stk.pop();
};